bee-api/bee-rest-api/src/types/dtos.rs

```diff
@@ -554,6 +707,476 @@ pub struct ReferenceUnlockBlockDto {
+            Self::ExpirationUnix(_) => ExpirationUnixFeatureBlock::KIND,
+            Self::Indexation(_) => IndexationFeatureBlock::KIND,
+            Self::Metadata(_) => MetadataFeatureBlock::KIND,
+        }
+    }
+}
+
+impl From<&FeatureBlock> for FeatureBlockDto {
+    fn from(value: &FeatureBlock) -> Self {
+        match value {
+            FeatureBlock::Sender(v) => Self::Sender(SenderFeatureBlockDto(v.address().into())),
+            FeatureBlock::Issuer(v) => Self::Issuer(IssuerFeatureBlockDto(v.address().into())),
+            FeatureBlock::DustDepositReturn(v) => Self::DustDepositReturn(DustDepositReturnFeatureBlockDto(v.amount())),
+            FeatureBlock::TimelockMilestoneIndex(v) => {
+                Self::TimelockMilestoneIndex(TimelockMilestoneIndexFeatureBlockDto(v.index()))
+            }
+            FeatureBlock::TimelockUnix(v) => Self::TimelockUnix(TimelockUnixFeatureBlockDto(v.timestamp())),
+            FeatureBlock::ExpirationMilestoneIndex(v) => {
+                Self::ExpirationMilestoneIndex(ExpirationMilestoneIndexFeatureBlockDto(v.index()))
+            }
+            FeatureBlock::ExpirationUnix(v) => Self::ExpirationUnix(ExpirationUnixFeatureBlockDto(v.timestamp())),
+            FeatureBlock::Indexation(v) => Self::Indexation(IndexationFeatureBlockDto(v.to_string())),
+            FeatureBlock::Metadata(v) => Self::Metadata(MetadataFeatureBlockDto(v.to_string())),
+        }
+    }
+}
+
+impl TryFrom<&FeatureBlockDto> for FeatureBlock {
+    type Error = Error;
+
+    fn try_from(value: &FeatureBlockDto) -> Result<Self, Self::Error> {
+        Ok(match value {
+            FeatureBlockDto::Sender(v) => Self::Sender(SenderFeatureBlock::new((&v.0).try_into()?)),
+            FeatureBlockDto::Issuer(v) => Self::Issuer(IssuerFeatureBlock::new((&v.0).try_into()?)),
+            FeatureBlockDto::DustDepositReturn(v) => Self::DustDepositReturn(DustDepositReturnFeatureBlock::new(v.0)?),
+            FeatureBlockDto::TimelockMilestoneIndex(v) => {
+                Self::TimelockMilestoneIndex(TimelockMilestoneIndexFeatureBlock::new(v.0))
+            }
+            FeatureBlockDto::TimelockUnix(v) => Self::TimelockUnix(TimelockUnixFeatureBlock::new(v.0)),
+            FeatureBlockDto::ExpirationMilestoneIndex(v) => {
+                Self::ExpirationMilestoneIndex(ExpirationMilestoneIndexFeatureBlock::new(v.0))
+            }
+            FeatureBlockDto::ExpirationUnix(v) => Self::ExpirationUnix(ExpirationUnixFeatureBlock::new(v.0)),
+            FeatureBlockDto::Indexation(v) => Self::Indexation(IndexationFeatureBlock::new(
+                &hex::decode(&v.0).map_err(|_e| Error::InvalidSemanticField("IndexationFeatureBlock"))?,
+            )?),
+            FeatureBlockDto::Metadata(v) => Self::Metadata(MetadataFeatureBlock::new(
+                &hex::decode(&v.0).map_err(|_e| Error::InvalidSemanticField("MetadataFeatureBlock"))?,
```

Comment:
like here, this decoded hex buffer is passed by reference but nobody else uses it. I assume `MetadataFeatureBlock::new` will copy some/all of the bytes instead of moving them.